### DISCUSSION TITLE
[Feature] "Include path in scan for new images" option per path in library

### DISCUSSION THREAD

**Discussion by v411e** [immich-app/immich#4274] [Feature Request]
### The feature

I have a lot of pictures in my external library. My folder structure is like this:

```
.
└── Camera 1
    ├── 2015
    ├── 2016
    ├── 2018
    ├── 2019
    ├── 2020
    ├── 2021
    ├── 2022
    └── 2023
        ├── 2023_01_01
        ├── 2023_01_12
        ├── ...
        └── 2023_12_30
```

Every time I add new pictures, they are most likely added into the current year folder. But when Immich scans my library for new images, it loops through all tens of thousands of files. This takes a long time and is not efficient at all.
Imagine adding 50 new Photos and 100.000 photos are being scanned.

As I have added each year as a separate path to my external library in Immich, it would be very handy for me to just _flag_ the folders that are unlikely to have new images to be excluded from scans. This might be combined with an option to trigger a scan on a specific path manually.

What do you think? Does anyone have the same problem or a better idea?

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by jrasm91** [comment]
There are two parts to this process:
1. Getting a list of file paths from the operating system
2. Sorting through the files and figuring out what is new

(1) is reasonably fast from what I understand, even with 100k files.
(2) is _currently_ very slow and inefficient. It does multiple nested for loops to find the "new" files and process them. 

(2) at least is being worked on in #4248 and should hopefully solve the issue you are facing.

**Reply by v411e** [reply]
Uh I was not aware of your PR. Thanks for working on this :)
